data_model/src/block.rs

```diff
@@ -132,31 +153,35 @@ impl BlockHeader {
     }
 }
 
-impl SignedBlockV1 {
+impl BlockPayload {
     /// Create new signed block, using `key_pair` to sign `payload`
     #[cfg(feature = "transparent_api")]
-    pub fn new(payload: BlockPayload, key_pair: &iroha_crypto::KeyPair) -> SignedBlockV1 {
-        let signature = iroha_crypto::SignatureOf::new(key_pair, &payload);
-        let signatures = SignaturesOf::from(signature);
+    pub fn sign(self, private_key: &iroha_crypto::PrivateKey) -> SignedBlock {
+        let signatures = vec![BlockSignature(0, SignatureOf::new(private_key, &self))];
+
         SignedBlockV1 {
             signatures,
-            payload,
+            payload: self,
         }
+        .into()
     }
+}
 
-    /// Block payload. Used for tests
-    #[cfg(feature = "transparent_api")]
-    pub fn payload(&self) -> &BlockPayload {
-        &self.payload
-    }
-
+impl SignedBlockV1 {
     #[cfg(feature = "std")]
     fn hash(&self) -> iroha_crypto::HashOf<SignedBlock> {
         iroha_crypto::HashOf::from_untyped_unchecked(iroha_crypto::HashOf::new(self).into())
     }
 }
 
 impl SignedBlock {
+    /// Block payload. Used for tests
+    #[cfg(feature = "transparent_api")]
```

Comment:
Why we put getters under `transparent_api` flag?

Reply:
just this one. Because `BlockPayload` is not supposed to be exposed